### PR TITLE
[python] build with ucs4

### PR DESCRIPTION
**Pull request by truthbk**

### CODE REVIEW

File: config/software/python.rb [hunk 1]
```diff
@@ -37,6 +37,7 @@
   }
 
   python_configure = ["./configure",
+                      "--enable-unicode=ucs4",
```
Comment (olivielpeau) [inline config/software/python.rb]: can we do this on Linux only? The default python on macOS is still built with `ucs2`, and the homebrew-provided python too, see https://github.com/Homebrew/homebrew-core/issues/12510

File: config/software/python.rb [hunk 2]
```diff
@@ -40,10 +40,12 @@
                       "--enable-universalsdk=/",
                       "--prefix=#{install_dir}/embedded"]
 
-  if ohai["platform_family"] == "mac_os_x"
+  if mac_os_x?
```
Comment (olivielpeau) [inline config/software/python.rb]: should be `osx?` I think: https://github.com/DataDog/dd-agent-omnibus/blob/5.22.3/lib/ostools.rb#L16-L18 & https://github.com/DataDog/datadog-agent/blob/6.2.1/omnibus/lib/ostools.rb#L20-L22

Comment (truthbk) [reply config/software/python.rb]: we have this in multiple recipes though...

Comment (olivielpeau) [reply config/software/python.rb]: ok this one is actually provided by `chef-sugar`

Comment (olivielpeau) [reply config/software/python.rb]: in the future we should probably get rid of our `lib/ostools.rb` and use the sugar-provided stuff directly instead

**Pull request merged**